### ISSUE TITLE
Add usage of the serial_test crate

### ISSUE THREAD

**Issue by volks73** [volks73/cargo-wix#131]
Currently, to run the tests, the `--test-threads=1` option must be used with the `cargo test` command because many of the integration tests alter the environment and execute from the current working directory. However, the `--test-threads=1` is a brute force method to have _all_ tests execute serially. Only the tests that need to execute serially would greatly speed up running all tests and it would eliminate the need for the `--test-threads=1` and caveat.

As of Rust v1.39, the [serial_test](https://crates.io/crates/serial_test) crate allows annotation of specific tests to be executed serially.

**Comment by volks73** [comment]
Maybe instead of the [serial_test](https://crates.io/crates/serial_test) crate, the [rusty-fork](https://crates.io/crates/rusty-fork) crate could be used?

The rusty-fork crate appears to create an isolated subprocess for each test inside the macro. This means each test that changes the current working directory (CWD) could be executed in parallel instead of serially. Each test would have its own CWD. This could greatly speed up running the tests, where as using the serial-test crate is more of a convenience for the developer to avoid having to remember to use `--test-threads=1`.

**Comment by volks73** [comment]
I have experimented with the [rusty-fork](https://crates.io/crates/rusty-fork) crate. It appears to work for creating isolated environments for each test, such that any test that modifies the Current Working Directory (CWD) can not be run in parallel. While this is great and does speed up running all of the integration tests, the various testing environment variables appear to no longer work. I have posted a question as an issue to the crate for some direction (https://github.com/AltSysrq/rusty-fork/issues/13). I am going to continue experimenting and working on this in the mean time.

**Comment by volks73** [comment]
I got impatient and decided to implement the serial_test crate after all. This eliminates the need for the `--test-threads=1` workaround. Completed as of 71b2725ca32f21403f3d27c95fb8ab6425af2b68.

I am going to create a separate issue to track usage of the rusty-fork crate instead of the serial_test crate.
